RFC-0006-nan-definition.md

```diff
@@ -0,0 +1,213 @@
+
+From the SQL:2016 spec:
+				
+					
+> Approximate numeric and decimal floating-point values consist of a mantissa and an exponent. The mantissa is a signed numeric value, and the exponent is a signed integer that specifies the magnitude of the mantissa. Approximate numeric and decimal floating-point values have a precision. The precision of approximate numeric values is a positive integer that specifies the number of significant binary digits in the mantissa. The precision of decimal floating-point values is a positive integer that specifies the number of significant decimal digits in the mantissa. The value of an approximate numeric or decimal floating-point type is the mantissa multiplied by a factor determined by the exponent.				
+An <exact numeric literal> ENL consists of either a <period> followed by an <unsigned integer> or an <unsigned integer> followed by an optional <period> and an optional <unsigned integer>. The declared type of ENL is an exact numeric type.				
+> An <approximate numeric literal> ANL consists of a <mantissa> that is an <exact numeric literal>, the letter 'E' or 'e', and an <exponent> that is a <signed integer>. It is implementation-defined whether the declared type of ANL is an approximate numeric type or the decimal floating-point type. If M is the value of the <mantissa>					
+and E is the value of the <exponent>, then M * 10E is the apparent value of ANL. If the declared type of ANL is an approximate numeric type, then the actual value of ANL is approximately the apparent value of ANL, according to implementation-defined rules. If the declared type of ANL is the decimal floating-point type, the actual value of ANL is either exactly or approximately the apparent value of ANL, according to implementation- defined rules. 
+>….							
+>Operations on numbers are performed according to the normal rules of arithmetic, within implementation-defined limits, except as provided for in Subclause 6.29, “<numeric value expression>”. 
+
+Some read the above as disallowing NaNs because floating point types are defined only as mantissa and exponent, and there are no special values defined like NaN, infinity, and -infinity.  I don't think it's clear one way or the other, and it looks like most databases do support NaN values.
+
+With regard to sorting, the ANSI spec says the following:							>											
+> f)  PVi is said to precede QVi if the value of the <comparison predicate> “PVi <comp op> QVi” is True for the applicable <comp op>.			
+>			 							
+> g)  If PVi and QVi are not the null value and the result of “PVi <comp op> QVi” is Unknown, then the relative ordering of PVi and QVi is implementation-dependent.
+
+### Current Presto behavior
+
+Here we give a short summary of Presto's current behavior.  There is a complete review here: https://docs.google.com/spreadsheets/d/1KclsskH88CLHMiu_Q2P25S3QfPc9H3Tu7h_9MXP10Dk/edit#gid=128250498  
+
+#### Comparison 
+
+= returns false
```

Comment:
It's not totally clear that you're talking about NaN = NaN and NaN <> NaN. There are also the cases where NaN = 7.2 or NaN <> 56.0

Reply:
All of the above. Any equality comparison will return false including NaN=NaN and any inequality comparison with another number will return true including NaN <> NaN.  I'll specify NaN = NaN and NaN <> NaN here because i think the other cases would be understood from that.